lld/COFF/Driver.cpp

```diff
@@ -657,7 +656,7 @@ void LinkerDriver::setMachine(MachineTypes machine) {
 
   ctx.config.machine = machine;
 
-  if (machine != ARM64X) {
+  if (!isArm64EC(machine)) {
     ctx.symtab.machine = machine;
   } else {
     ctx.symtab.machine = ARM64EC;
```

Comment:
It would probably be good to add a comment here.

Reply:
I added a comment, thanks for review!